### PR TITLE
Add `j.u.s.Stream#forEach`

### PR DESCRIPTION
**Pull request by armanbilge**
Targeting 0.4.x.

### CODE REVIEW

File: unit-tests/shared/src/test/scala/javalib/util/stream/StreamTest.scala [hunk 1]
```diff
@@ -46,6 +46,13 @@ class StreamTest {
     assertFalse(it.hasNext())
   }
 
+  @Test def streamForeachWorks(): Unit = {
+    val s = Stream.of(1, 2, 3)
+    var sum = 0
+    s.forEach(sum += _)
```
Comment (sideeffffect) [inline unit-tests/shared/src/test/scala/javalib/util/stream/StreamTest.scala:52]: This line seems to cause issues for Scala 2.11
```scala
[error] D:\a\scala-native\scala-native\unit-tests\shared\src\test\scala\javalib\util\stream\StreamTest.scala:52:22: missing parameter type for expanded function ((x$1) => sum.<$plus$eq: error>(x$1))
[error]     s.forEach(sum += _)
[error]                      ^
[error] one error found
[error] (testsJVM2_11 / Test / compileIncremental) Compilation failed
```
https://github.com/scala-native/scala-native/runs/7479078735?check_suite_focus=true#step%3A5%3A69=

**Review by sideeffffect — CHANGES_REQUESTED** [review]

File: unit-tests/shared/src/test/scala/javalib/util/stream/StreamTest.scala (hunk 1 above)
Comment (armanbilge) [reply unit-tests/shared/src/test/scala/javalib/util/stream/StreamTest.scala]: Thanks :)

**Comment by ekrich** [comment]
Is there anyway this can be done without delegating to Scala stdlib? We have been trying to decouple like Scala.js. https://github.com/scala-js/scala-js/pull/4677 ScalaOps can help your sanity in the process.

**Comment by armanbilge** [comment]
That's a great point. Unfortunately not in this PR, since it seems that the Java Stream is implemented in terms of the Scala one.

**Comment by ekrich** [comment]
If you are still working on this could you add a comment at the top as a reminder?

**Comment by armanbilge** [comment]
Seems more appropriate as an issue? Since it applies to the entire project, no? :)

**Comment by ekrich** [comment]
Yes, I think that is more appropriate. I searched and could not find one. We can keep track of items as we find them.

**Pull request merged**